packages/itwinui-react/src/core/utils/components/ShadowRoot.tsx

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+const isBrowser = typeof document !== 'undefined';
+const supportsDSD =
+  isBrowser && 'shadowRootMode' in HTMLTemplateElement.prototype;
+
+/**
+ * Wrapper around `<template>` element that attaches shadow root to its parent
+ * and moves its children into the shadow root.
+ */
```

Comment:
nit: Since this is a private util component, maybe we can mark it with `@private`.

```suggestion
 *
 * @private
 */
```